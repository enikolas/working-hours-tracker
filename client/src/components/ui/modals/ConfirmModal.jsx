import React from 'react';
import PropTypes from 'prop-types';

import Modal from '../Modal';
import strings from '../../../utils/strings';

const ConfirmModal = props => (
	<Modal
		active={props.active}
		title={props.title ? props.title : strings.confirmation}
		content={props.content}
		buttons={[
			{ action: props.onCancel, label: strings.no },
			{ action: props.onConfirm, label: strings.yes }
		]}
	/>
);

ConfirmModal.propTypes = {
	active: PropTypes.bool,
	title: PropTypes.string,
	content: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.node
	]).isRequired,
	onCancel: PropTypes.func,
	onConfirm: PropTypes.func.isRequired
};

ConfirmModal.defaultProps = {
	active: false,
	title: '',
	onCancel: () => {}
};

export default ConfirmModal;
