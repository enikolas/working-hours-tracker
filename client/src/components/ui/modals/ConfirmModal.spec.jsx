import React from 'react';
import { shallow } from 'enzyme';

import ConfirmModal from './ConfirmModal';

const propsContent = 'this is a test confirm modal';
const propsOnConfirm = jest.fn();

describe('ConfirmModal', () => {
	it('should render properly with the necessary props', () => {
		const wrapper = shallow(<ConfirmModal
			content={propsContent}
			onConfirm={propsOnConfirm}
		/>);
		expect(wrapper).toMatchSnapshot();
	});
	it('should render with a new title name', () => {
		const wrapper = shallow(<ConfirmModal
			title="this is a confirm modal title"
			content={propsContent}
			onConfirm={propsOnConfirm}
		/>);
		expect(wrapper).toMatchSnapshot();
	});
});
