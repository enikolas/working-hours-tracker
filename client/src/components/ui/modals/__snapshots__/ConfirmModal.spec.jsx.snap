// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal should render properly with the necessary props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmModal
    active={false}
    content="this is a test confirm modal"
    onCancel={[Function]}
    onConfirm={[MockFunction]}
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": false,
      "buttons": Array [
        Object {
          "action": [Function],
          "label": "Não",
        },
        Object {
          "action": [MockFunction],
          "label": "Sim",
        },
      ],
      "cancel": null,
      "content": "this is a test confirm modal",
      "hasButtons": true,
      "title": "Confirmação",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": false,
        "buttons": Array [
          Object {
            "action": [Function],
            "label": "Não",
          },
          Object {
            "action": [MockFunction],
            "label": "Sim",
          },
        ],
        "cancel": null,
        "content": "this is a test confirm modal",
        "hasButtons": true,
        "title": "Confirmação",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ConfirmModal should render with a new title name 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmModal
    active={false}
    content="this is a test confirm modal"
    onCancel={[Function]}
    onConfirm={[MockFunction]}
    title="this is a confirm modal title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": false,
      "buttons": Array [
        Object {
          "action": [Function],
          "label": "Não",
        },
        Object {
          "action": [MockFunction],
          "label": "Sim",
        },
      ],
      "cancel": null,
      "content": "this is a test confirm modal",
      "hasButtons": true,
      "title": "this is a confirm modal title",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": false,
        "buttons": Array [
          Object {
            "action": [Function],
            "label": "Não",
          },
          Object {
            "action": [MockFunction],
            "label": "Sim",
          },
        ],
        "cancel": null,
        "content": "this is a test confirm modal",
        "hasButtons": true,
        "title": "this is a confirm modal title",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
